src/hit/api/evaluator/homomorphic.cpp

```diff
@@ -338,11 +329,11 @@ namespace hit {
     }
 
     void HomomorphicEval::rescale_to_next_inplace_internal(CKKSCiphertext &ct) {
-        rescaleMany(get_evaluator(), context->params, ct.backend_ct, 1, ct.backend_ct);
+        rescaleMany(get_evaluator().ref(), context->params, ct.backend_ct, 1, ct.backend_ct);
     }
 
     void HomomorphicEval::relinearize_inplace_internal(CKKSCiphertext &ct) {
-        relinearize(get_evaluator(), ct.backend_ct, ct.backend_ct);
+        relinearize(get_evaluator().ref(), ct.backend_ct, ct.backend_ct);
```

Comment:
Why not have `get_evaluator()` call `.ref()`?

Reply:
I see: it's so that when `PoolObject` destructor is called so that we release the object back to the `ObjectPool`.

Reply:
Because `get_evaluator()` needs to return the tracking `PoolObject` object. This is the object which (in the destructor) returns the tracked object to the pool. If `get_evaluator()` were to call `.ref()` internally then I think that the destructor on the `PoolObject` would be called within `get_evaluator()` and thus the evaluator would be returned to the pool and eligible for use by another thread.